### PR TITLE
Cfy 3691 make all internal rest calls secure

### PR DESCRIPTION
**Pull request by noakup**

### CODE REVIEW

File: cloudify_agent/api/pm/base.py [hunk 1]
```diff
@@ -111,10 +110,22 @@ class Daemon(object):
...
 
         the manager REST gateway port to connect to. defaults to 8101.
 
+    ``manager_username``:
+
+        the username to use in REST call. No default.
+
+    ``manager_password``:
+
+        the password to use in REST call. No default.
```
Comment (dankilman) [inline cloudify_agent/api/pm/base.py]: add docs for certificate related

File: cloudify_agent/api/pm/base.py [hunk 2]
```diff
@@ -759,7 +780,15 @@ def _get_queue_from_manager(self):
         return self._runtime_properties['cloudify_agent']['queue']
 
     def _get_runtime_properties(self):
-        client = CloudifyClient(host=self.manager_ip, port=self.manager_port)
+        client = utils.get_rest_client(
```
Comment (dankilman) [inline cloudify_agent/api/pm/base.py]: propagate security_enabled somehow

File: cloudify_agent/api/pm/detach.py [hunk 3]
```diff
@@ -178,6 +178,10 @@ def _create_config(self):
             broker_url=self.broker_url,
             manager_ip=self.manager_ip,
             manager_port=self.manager_port,
+            manager_protocol=self.manager_protocol,
```
Comment (dankilman) [inline cloudify_agent/api/pm/detach.py]: add certificate related + enabled

File: setup.py [hunk 4]
```diff
@@ -25,7 +25,8 @@
     'jinja2==2.7.2',
     'pywinrm==0.0.3',
     'fabric==1.8.3',
-    'wagon==0.2.5'
+    'wagon==0.2.5',
+    'itsdangerous==0.24'
```
Comment (dankilman) [inline setup.py]: remove

File: cloudify_agent/api/utils.py [hunk 5]
```diff
@@ -161,10 +162,15 @@ def daemon_to_dict(daemon):
...
-            agent['manager_port'],
-        )
+        client = get_rest_client(
+            manager_ip=agent['manager_ip'],
+            manager_protocol=agent['manager_protocol'],
+            manager_port=agent['manager_port'],
+            cloudify_username=agent['manager_username'],
+            cloudify_password=agent['manager_password'],
+            verify_ssl_certificate=agent.get('verify_manager_certificate'),
+            ssl_cert_path=agent.get('local_manager_cert_path'))
```
Comment (dankilman) [inline cloudify_agent/api/utils.py]: add security_enabled

File: cloudify_agent/installer/__init__.py [hunk 6]
```diff
@@ -173,6 +173,16 @@ def _create_agent_env(self):
             env.CLOUDIFY_BROKER_IP: self.cloudify_agent.get('broker_ip'),
             env.CLOUDIFY_BROKER_PORT: self.cloudify_agent.get('broker_port'),
             env.CLOUDIFY_MANAGER_PORT: self.cloudify_agent.get('manager_port'),
+            env.CLOUDIFY_MANAGER_PROTOCOL: self.cloudify_agent.get(
```
Comment (dankilman) [inline cloudify_agent/installer/__init__.py]: add security_enabled

File: cloudify_agent/installer/config/attributes.py [hunk 7]
```diff
@@ -66,8 +65,22 @@
         'group': 'cfy-agent'
     },
     'manager_port': {
-        'group': 'cfy-agent',
-        'default': defaults.MANAGER_PORT
+        'group': 'cfy-agent'
```
Comment (dankilman) [inline cloudify_agent/installer/config/attributes.py]: security_enabled

File: cloudify_agent/installer/config/configuration.py [hunk 8]
```diff
@@ -165,6 +171,17 @@ def _cfy_agent_attributes_no_defaults(cloudify_agent):
         # by default, the manager ip will be set by an environment variable
         cloudify_agent['manager_ip'] = get_manager_ip()
 
+    security_enabled = ctx.security_ctx.security_enabled
```
Comment (dankilman) [inline cloudify_agent/installer/config/configuration.py]: 1) validate if security_enabled and not (username and password)
2) add `security_enabled` + certificate related

File: cloudify_agent/resources/pm/detach/detach.conf.template [hunk 9]
```diff
@@ -6,6 +6,11 @@ export CLOUDIFY_DAEMON_STORAGE_DIRECTORY={{ storage_dir }}
 export CLOUDIFY_DAEMON_USER={{ user }}
 export MANAGEMENT_IP={{ manager_ip }}
 export MANAGER_REST_PORT={{ manager_port }}
+export MANAGER_REST_PROTOCOL={{ manager_protocol }}
```
Comment (dankilman) [inline cloudify_agent/resources/pm/detach/detach.conf.template]: add certificate related + enabled

File: cloudify_agent/shell/commands/daemons.py [hunk 10]
```diff
@@ -40,6 +40,28 @@
               help='The manager REST gateway port to connect to. [env {0}]'
               .format(env.CLOUDIFY_MANAGER_PORT),
               envvar=env.CLOUDIFY_MANAGER_PORT)
+@click.option('--manager-protocol',
```
Comment (dankilman) [inline cloudify_agent/shell/commands/daemons.py]: add security_enabled

File: cloudify_agent/shell/commands/installer.py [hunk 11]
```diff
@@ -34,7 +34,9 @@ def install_local(agent_file, output_agent_file):
     if agent_file is None:
         raise click.ClickException('--agent-file should be specified.')
     cloudify_agent = json.load(agent_file)
-    cloudify_agent['manager_port'] = defaults.MANAGER_PORT
+    # TODO should not always take the default,
```
Comment (dankilman) [inline cloudify_agent/shell/commands/installer.py]: consult with Konrad regarding agent migration paths

File: cloudify_agent/shell/env.py [hunk 12]
```diff
@@ -26,6 +26,11 @@
 CLOUDIFY_BROKER_IP = 'CLOUDIFY_BROKER_IP'
 CLOUDIFY_BROKER_PORT = 'CLOUDIFY_BROKER_PORT'
 CLOUDIFY_MANAGER_PORT = 'CLOUDIFY_MANAGER_PORT'
+CLOUDIFY_MANAGER_PROTOCOL = 'CLOUDIFY_MANAGER_PROTOCOL'
```
Comment (dankilman) [inline cloudify_agent/shell/env.py]: add security_enabled

**Comment by noakup** [comment]
This PR seems like a duplicate of PR #62. Closing the PR but not deleting the branch until #62 is merged.

**Pull request closed without merging**